plasmapy/tests/helpers/exceptions.py

```diff
@@ -0,0 +1,109 @@
+"""Exceptions that describe different types of test failures."""
+
+__all__ = [
+    "Failed",
+    "UnexpectedResultError",
+    "InconsistentTypeError",
+    "MissingExceptionError",
+    "UnexpectedExceptionError",
+    "MissingExceptionError",
+    "UnexpectedWarningError",
+    "MissingWarningError",
+    "InvalidTestError",
+    "ExceptionMismatchError",
+    "WarningMismatchError",
+]
+
+import pytest
+
+# TODO: get the error message while raising this to be simply "Failed"
+#       rather than plasmapy.tests.helpers.exceptions.Failed
+
+
+class Failed(pytest.fail.Exception):
+    """
+    Base exception for test failures.
+
+    Notes
+    -----
+    This exception was derived from `~pytest.fail.Exception`, which in
+    turn was derived from `BaseException` (not `Exception`).
+    """
+
+    pass
+
+
+class InconsistentTypeError(Failed):
```

Comment:
Should this also inherit from `TypeError`, to be consistent?

Reply:
Despite the name, this is different from a `TypeError`.  A `TypeError` is to be (from Python 3 docs): "raised when an operation or function is applied to an object of inappropriate type"...which is not happening here.  In this case, the types of two objects are being checked against each other, and are found to not be the same.  I wonder if we should rename this to something like `InconsistentTypes`...

Reply:
Maybe instead of calling these `<something>Error` we should call them `<something>Failure`.  Since this is really a failure opposed to an error.  For reference, `pytest.fail` does...

```python
>>> pytest.fail()
Failed: <Failed instance>
```